compiler/test-resources/repl/i6676

```diff
@@ -7,8 +7,8 @@ scala> xml"
   |     ';' expected, but eof found
 scala> xml""
 1 | xml""
-  |    ^
-  |    value xml is not a member of StringContext
+  |     ^
```

Comment:
This is now aligned with the implementation, but it still looks wrong to me. Analogous to

```
scala> "".xml
1 |"".xml
  |^^^^^^
  |value xml is not a member of String
```

I would have expected

```
scala> "".xml
1 |xml""
  |^^^^^
  |value xml is not a member of StringContext
```

Reply:
Indeed, that would be the correct position.

Change [Desugar.scala#L1668](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/ast/Desugar.scala#L1668) to
```scala
        Apply(Select(Apply(scalaDot(nme.StringContext).withSpan(tree.span), strs), id), elems)
```